### ISSUE TITLE
add an argument to drop incomplete factor combinations?

### ISSUE THREAD

**Issue by ha0ye** [weecology/portalr#88]
The current output for abundance at `level = "treatment"` has:
```
# A tibble: 1,468 x 23
   period treatment    BA    DM    DO    DS  `NA`    OL    OT    PB    PE    PF    PH    PI
 *  <dbl>    <fctr> <int> <int> <int> <int> <int> <int> <int> <int> <int> <int> <int> <int>
 1      1   control     0    42     0     6     4     0     1     0     2     3     0     0
 2      2   control     0    63     1    13     4     0     0     0     0     8     0     0
 3      3   control     0    45     0    24     4     0     0     0     2     5     0     0
 4      4   control     0    40     3    19     6     3     3     0     2     1     0     0
```
whereas, for abundance at `level = "plot"` has:
```
# A tibble: 30 x 23
   period  plot    BA    DM    DO    DS  `NA`    OL    OT    PB    PE    PF    PH    PI    PL
    <dbl> <int> <int> <int> <int> <int> <int> <int> <int> <int> <int> <int> <int> <int> <int>
 1      1     1     0     2     0     0     0     0     0     0     0     1     0     0     0
 2      1     2     0     1     0     0     1     0     0     0     1     0     0     0     0
 3      1     3     0     2     0     1     1     0     0     0     0     0     0     0     0
 ...
23      1    23     0     3     0     0     0     0     0     0     0     0     0     0     0
24      1    24    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA
```

The first table drops missing levels of period x treatment, the second table includes the unsampled period x plot levels, but sets abundance to NA.

On solution for consistency is to add an argument to allow missing factor combinations to be dropped or included in the output (they'll probably always have to be set to NA for the cross-table output format).

**Comment by gmyenni** [comment]
This will require some coordination with the current `incomplete` argument, which removes any trapping sessions where >0 but <24 plots were trapped. It's a part of the reason why the NAs are inconsistent. 
It seems like we do need to have two arguments to get all this done:

1. An `incomplete` argument that has options:
    1. remove all incomplete sampling
    2. use all raw abundances 
    3. adjust abundances according to number of plots trapped (#47 )

2. An `na.fill` argument that works after `incomplete` and has options:
    1. leave out missing combinations
    2. fill missing combinations with NA

In this case, it should only work on combinations of period x treatment, not including species. So it should be consistent across table formats?

**Comment by ha0ye** [comment]
That makes sense to me. Discuss at next week's meeting?

**Comment by ha0ye** [comment]
@diazrenata and I will meet to discuss this in detail

**Comment by gmyenni** [comment]
Closed via #114
